### ISSUE TITLE
UnsupportedPythonVersion with two Pipfile [[source]] blocks

### ISSUE THREAD

**Issue by commandtab** [pypa/pipenv#1179]
When installing `python-swiftclient` from a Pipfile with a single `[[source]]` block, like so:

```toml
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"

[packages]
python-swiftclient = {version="==3.4.0", index="pypi"}
```

`python-swiftclient` installs successfully. So far so good. But, when my Pipfile has two valid `[[source]]` blocks...

```toml
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"

[[source]]
url = "https://pypi.companyname.com"
verify_ssl = true
name = "privatepypi"

[packages]
python-swiftclient = {version="==3.4.0", index="pypi"}
```

Installation of `python-swiftclient` errors with...

```
pip.exceptions.UnsupportedPythonVersion: futures requires
Python '>=2.6, <3' but the running Python is 3.6.3
```

...despite explicitly specifying which index `python-swiftclient` should be installed from (`index="pypi"`). Why does having a second (and theoretically unused) `[[source]]` block affect installation of that package? (Of note, `privatepypi` does not contain a fork of `python-swiftclient`.)

##### Describe your environment

1. OS Type: macOS Sierra 10.12.6
1. Python version: 3.6.3
1. Pipenv version: 8.3.2

##### Expected result

I expected `python-swiftclient` to install from public pypi without issue when two `[[source]]` blocks are present and the correct index to use is specified for the given package.

##### Actual result

`python-swiftclient` fails to install when two `[[source]]` blocks are present, suggesting that the un-referenced `[[source]]` is being consulted or is affecting dependency resolution somehow.

##### Steps to replicate

Write this to a `Pipfile`:

```toml
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"

[[source]]
url = "https://pypi.companyname.com"
verify_ssl = true
name = "privatepypi"

[packages]
python-swiftclient = {version="==3.4.0", index="pypi"}
```

Then do `pipenv --three install` to observe the error:

```
pip.exceptions.UnsupportedPythonVersion: futures 
requires Python '>=2.6, <3' but the running Python is 3.6.3
```

Any suggestions? Thanks!

**Comment by techalchemy** [comment]
@commandtab I actually can't reproduce this with your pipfile which means one of a few things:

1. You may have pipenv installed via pip on python 2,7, which could be causing a problem here.  I'd recommend installing pipenv via pip3 and using that
2. You may be running into a caching issue, which could be resolved via `pipenv lock --clear`

Having played around a tiny bit with this I think the first case is the most likely one, we have a known (and complicated) dependency resolution issue when crossing from python 2.7 installed pipenv into python 3. Can you give us the output of `pipenv lock --verbose --clear` before you reinstall pipenv with pip3?

Thanks for the report!

/cc @vphilippon

**Comment by commandtab** [comment]
Thanks so much! You're right on with 1!

I have Python 2 and 3 installed via Homebrew because I maintain projects written against both versions, and pipenv is installed under Python 2.7.14 (it was the default `pip` when doing `pip install pipenv`). However, this particular project in question is a Python 3.6.3 one. Here is the output of `pipenv lock --verbose --clear` under this arrangement:

https://gist.github.com/commandtab/9b88410e02aff38fd8b70d5bd2ba7b92

After uninstalling pipenv using Python 2's pip and reinstalling it with Python 3's pip, the same command successfully installs `python-swiftclient` 🎉   Here's the output for that, just in case:

https://gist.github.com/commandtab/cfbc1db2956c26dd92492a3d8f08499e

Now part of me is wondering if the reverse will run into a similar problem, where pipenv is installed in a 3.x environment and is installing packages for a 2.7.x project... 😅 

Thanks again for your help! I truly appreciate it.

**Comment by techalchemy** [comment]
@commandtab sure thing, we see many more issues going from python 2 installed pipenv to python 3 environments than the other way around.  Pipenv on python 3 works pretty consistently in my own experience.

As a sidenote after I commented yesterday I did manage to run the same script and install a python 3 environment using a python 2 installation of pipenv via pyenv using pipenv 9.0, but I will note that it failed once and took quite awhile.  I'll close this for now since we are tracking this in a few places but I'll make note of the lock output since that is what is relevant to solving the dependency issue.  Thanks for helping us troubleshoot!
